static/js/filter.js

```diff
@@ -82,7 +82,12 @@ function message_matches_search_term(message, operator, operand) {
         if (page_params.realm_is_zephyr_mirror_realm) {
             return zephyr_stream_name_match(message, operand);
         }
-        return (message.stream.toLowerCase() === operand);
+        var stream_id = stream_data.get_stream_id(operand);
+        if (!stream_id) {
+            return false;
+        }
+
+        return (message.stream_id === stream_id);
```

Comment:
I guess the other option is to make that `return false` line instead be the old `return (message.stream.toLowerCase() === operand);`.  The logic would only run if no stream with that name exists, which is probably fairly precisely the case in question?